internal/provider/archetype_data_source.go

```diff
@@ -689,23 +689,38 @@ func policyAssignmentType2ArmPolicyAssignment(pamap map[string]PolicyAssignmentT
 		dst.Type = to.Ptr(policyAssignementType)
 		dst.Properties.DisplayName = to.Ptr(src.DisplayName.ValueString())
 
+		policyDefinitionMode := "NotSupported"
```

Comment:
I wonder if we should use a stronger type than a string here. Thinking of iota or type/const based approach.

Reply:
I am having second thoughts about this change anyway. Can discuss next week. I think the module fix will suffice for now.